### PR TITLE
Adds E0000212 .phishingAttemptDetected ServerErrorCode

### PR DESCRIPTION
**Pull request by michaelbiviano-okta**
### Problem Analysis (Technical)

Add support for phishing attempt detection

### Solution (Technical)

Add `ServerErrorCode` E0000212 to support phishing attempt detection

### Affected Components

ServerErrorCode

### Steps to reproduce:

Actual result: N/A

Expected result:  N/A

### Tests

Tests/DeviceAuthenticatorUnitTests/AuthenticatorEnrollmentTests.swift
Tests/DeviceAuthenticatorUnitTests/OktaSharedSQLiteTests.swift
Tests/DeviceAuthenticatorUnitTests/ServerErrorCodesTests.swift

### CODE REVIEW

File: Sources/DeviceAuthenticator/Enrollment/AuthenticatorEnrollment.swift [hunk 1]
```diff
@@ -241,7 +241,7 @@ class AuthenticatorEnrollment: AuthenticatorEnrollmentProtocol {
 
     func state(from errorCode: ServerErrorCode) -> EnrollmentState {
         switch errorCode {
-        case .userDeleted, .enrollmentNotFound:
+        case .userDeleted, .enrollmentNotFound, .phishingAttemptDetected:
```
Comment (IldarAbdullin-okta) [inline Sources/DeviceAuthenticator/Enrollment/AuthenticatorEnrollment.swift:244]: Can we return current known enrollment state for this error? phishingAttemptDetected is transactional error, it is not related to enrollment lifecycle

**Review by IldarAbdullin-okta — CHANGES_REQUESTED** [review]
Please address comment

File: Sources/DeviceAuthenticator/Enrollment/AuthenticatorEnrollment.swift (hunk 1 above)
Comment (michaelbiviano-okta) [reply Sources/DeviceAuthenticator/Enrollment/AuthenticatorEnrollment.swift]: Updated

File: Sources/DeviceAuthenticator/Enrollment/AuthenticatorEnrollment.swift [hunk 2]
```diff
@@ -253,6 +253,8 @@ class AuthenticatorEnrollment: AuthenticatorEnrollmentProtocol {
             return .reset
         case .unknown, .invalidToken:
             return .active
+        case .phishingAttemptDetected:
+            return state
```
Comment (IldarAbdullin-okta) [inline Sources/DeviceAuthenticator/Enrollment/AuthenticatorEnrollment.swift:257]: 👍

**Review by IldarAbdullin-okta — APPROVED** [review]
LGTM!

**Review by nkeerthy-okta — APPROVED** [review]
👍

**Pull request merged**